python/lsst/daf/persistence/safeFileIo.py

```diff
@@ -48,6 +50,40 @@ def setFileMode(filename):
     os.chmod(filename, (~umask & 0o666))
 
 @contextmanager
+def FileForWriteOnceCompareSame(name):
+    """Context manager to get a file that can be written only once and all other writes will succeed only if
+    they match the inital write.
+
+    The context manager provides a temporary file object. After the user is done, the temporary file becomes
+    the permanant file if the file at name does not already exist. If the file at name does exist the 
+    temporary file is compared to the file at name. If they are the same then this is good and the temp file
+    is silently thrown away. If they are not the same then a runtime error is raised.
+    """
+    outDir, outName = os.path.split(name)
+    safeMakeDir(outDir)
+    with tempfile.NamedTemporaryFile(dir=outDir, prefix=outName, delete=False) as temp:
+        try: 
+            yield temp
+        finally:
+            try:
+                temp.close()
```

Comment:
Could this plus the `with` statement lead to two `close()` calls?  Is there any chance that could be unsafe?